packages/next-swc/crates/next-core/src/next_pages/page_entry.rs

```diff
@@ -34,116 +38,83 @@ pub async fn create_page_ssr_entry_module(
-                    export const reportWebVitals = hoist(userland, "reportWebVitals")
-
-                    export const unstable_getStaticProps = hoist(userland, "unstable_getStaticProps")
-                    export const unstable_getStaticPaths = hoist(userland, "unstable_getStaticPaths")
-                    export const unstable_getStaticParams = hoist(userland, "unstable_getStaticParams")
-                    export const unstable_getServerProps = hoist(userland, "unstable_getServerProps")
-                    export const unstable_getServerSideProps = hoist(userland, "unstable_getServerSideProps")
-                    
-                    export const routeModule = new RouteModule({{
-                        definition: {{
-                            kind: "PAGES",
-                            page: {definition_page},
-                            pathname: {definition_pathname},
-                            // The following aren't used in production, but are
-                            // required for the RouteModule constructor.
-                            bundlePath: "",
-                            filename: "",
-                        }},
-                        components: {{
-                            App,
-                            Document,
-                        }},
-                        userland,
-                    }})
-                "#,
-                definition_page = StringifyJs(&definition_page),
-                definition_pathname = StringifyJs(&definition_pathname),
-            )?;
-
-            // When we're building the instrumentation page (only when the
-            // instrumentation file conflicts with a page also labeled
-            // /instrumentation) hoist the `register` method.
-            if definition_page.as_str() == "/instrumentation"
-                || definition_page.as_str() == "/src/instrumentation"
-            {
-                writeln!(
-                    result,
-                    r#"export const register = hoist(userland, "register")"#
-                )?;
-            }
-
-            "pages-entry.tsx".to_string()
+            // Load the Page entry file.
+            "/dist/esm/build/webpack/loaders/next-route-loader/entries/pages.js"
         }
         ReferenceType::Entry(EntryReferenceSubType::PagesApi) => {
-            // Sourced from https://github.com/vercel/next.js/blob/2848ce51d1552633119c89ab49ff7fe2e4e91c91/packages/next/src/build/webpack/loaders/next-route-loader/index.ts
-            writedoc!(
+            // Load the Pages API entry file.
+            "/dist/esm/build/webpack/loaders/next-route-loader/entries/pages-api.js"
+        }
+        _ => bail!("Invalid path type"),
+    };
+
+    // Load the file from the next.js codebase.
+    let file = load_next_js(project_root, template_file).await?.await?;
+
+    let mut file = file
+        .to_str()?
+        .replace(
+            "\"VAR_DEFINITION_PAGE\"",
```

Comment:
There is a `regex` crate, if you want to port over your code and (importantly) invariants